### ISSUE TITLE
Char loading fails for large charts on RPi 3

### ISSUE THREAD

**Issue by RBerliner** [SignalK/signalk-java#5]
I have discovered that Chartloading fails for the RPI 3 when the zip file is large ( > 100 MB) but works OK when the file size is small (< 10 MB).

I now report that when running SignalK on my Lubuntu laptop, the large files will load OK. This suggests that the problem is connected to the RPi execution and not the SignalK code. Out of memory?

Ron

**Comment by rob42** [comment]
Yes, quite likely. But it still needs to work, so needs attention. Ive looked at the code, it should be streaming to temp file, then copying, not staying in RAM. I need to look again.

Rob

November 27 2016 3:27 PM, "Ron Berliner"  wrote:
	I have discovered that Chartloading fails for the RPI 3 when the zip file is large ( > 100 MB) but works OK when the file size is small (< 10 MB).

	I now report that when running SignalK on my Lubuntu laptop, the large files will load OK. This suggests that the problem is connected to the RPi execution and not the SignalK code. Out of memory?

	Ron

**Comment by RBerliner** [comment]
I believe that I have found a solution to the problem of large chart loading failures.

I added the line:

export MAVEN_OPTS="-Xms1024m -Xmx1024m -XX:PermSize=1024m"

to the bottom of .bashrc on the RPi according to the instructions here: http://docs.alfresco.com/4.2/tasks/dev-extensions-maven-sdk-maven-opts.html

SignalK uses the same Java VM and options as mvn when run from "mvn exec:java"

On my RPi 3, the command free gives:

pi@pi:~ $ free
             total       used       free     shared    buffers     cached
Mem:        947740     305752     641988       8608       5976     201328
-/+ buffers/cache:      98448     849292
Swap:       102396      18020      84376

I am using a 32 GB SD card. 

When mvn exec:java starts up it complains that PermSize support was removed but otherwise proceeds.

"Java HotSpot(TM) Client VM warning: ignoring option PermSize=1024m; support was removed in 8.0"

I was then able to load NOAA 11552_1 (107 MBytes) which failed before. It displays too!
